api/app.py

```diff
@@ -42,25 +48,63 @@
 PalaceXrayProfiler.configure(app)
 
 
-@app.before_first_request
-def initialize_database(autoinitialize=True):
-    testing = "TESTING" in os.environ
+def initialize_application() -> Flask:
+
+    initialize_database()
+    initialize_circulation_manager()
+    initialize_admin()
```

Comment:
I think in order to make sure that the DB session we are opening here is cleaned up correctly, we need to do something like this: 

```python
def initialize_application() -> Flask:
    with app.application_context():
        initialize_database()
        initialize_circulation_manager()
        initialize_admin()
```

Reply:
Good catch.  I will fix.